mantle/kola/harness.go

```diff
@@ -670,17 +699,37 @@ func runProvidedTests(testsBank map[string]*register.Test, patterns []string, mu
 	return firstRunErr
 }
 
+func GetRerunnableTestName(testName string) (string, bool) {
+	// The current nonexclusive test wrapper would rerun all non-exclusive tests.
+	// Instead, we only want to rerun the one(s) that failed, so we will not consider
+	// the wrapper as "rerunnable".
+	match := nonexclusiveMatch.MatchString(testName)
+	// Test is not rerunnable if the test name matches the wrapper pattern
+	if match {
+		return "", false
+	} else {
```

Comment:
the else clause here feels weird. We could just unindent it since the true part of the condition is a return. We could even just make the true part embedded above like:


```code
if match := nonexclusiveMatch.MatchString(testName); match {
	// Test is not rerunnable if the test name matches the wrapper pattern
    return "", false
}
```

or would this even work?

```code
if nonexclusiveMatch.MatchString(testName) {
	// Test is not rerunnable if the test name matches the wrapper pattern
    return "", false
}
```

Reply:
I wanted to put this comment `// The current nonexclusive test wrapper would rerun all ...`, above the regex match statement for context. I can change it if its confusing :+1: